src/nspawn/nspawn.c

```diff
@@ -3435,6 +3435,12 @@ static int inner_child(
         if (arg_selinux_context)
                 if (setexeccon(arg_selinux_context) < 0)
                         return log_error_errno(errno, "setexeccon(\"%s\") failed: %m", arg_selinux_context);
+        if (arg_selinux_apifs_context) {
+                if ((setfilecon("/dev/mqueue", arg_selinux_apifs_context) < 0) && errno != ENOENT)
+                        return log_error_errno(errno, "setffilecon(\"/dev/mqueue\",\"%s\") failed: %m", arg_selinux_context);
+                if ((setfilecon("/dev/hugepages", arg_selinux_apifs_context) < 0) && errno != ENOENT)
+                        return log_error_errno(errno, "setffilecon(\"/dev/hugepages\",\"%s\") failed: %m", arg_selinux_context);
+        }
```

Comment:
hmm, we do not create /dev/hugepages at all, this should not be necessary

Reply:
Ok this is done via a unit file I guess within the environment.